packages/kit/src/runtime/client/renderer.js

```diff
@@ -596,19 +591,15 @@ export class Renderer {
 		const [pattern, a, b, get_params] = route;
 		const params = get_params
 			? // the pattern is for the route which we've already matched to this path
-			  get_params(/** @type {RegExpExecArray}  */ (pattern.exec(decoded_path)))
+			  get_params(/** @type {RegExpExecArray}  */ (pattern.exec(url.pathname))) // TODO this needs to account for paths.base
```

Comment:
`pathname` is not decoded and so I think this will break when you try to do the route matching. I'm not sure you can discard `decoded_path`. I'd be hesitant to change this as it's been a big source of bugs. The server-side is pretty solid at this point, but I don't know if I wrote as many tests for the client-side

Reply:
fixed this and added a test